graph/src/components/store.rs

```diff
@@ -393,19 +415,255 @@ pub trait Store: Send + Sync + 'static {
 
     /// Counts the total number of entities in a subgraph.
     fn count_entities(&self, subgraph: SubgraphDeploymentId) -> Result<u64, Error>;
-}
 
-pub trait SubgraphDeploymentStore: Send + Sync + 'static {
     fn resolve_subgraph_name_to_id(
         &self,
         name: SubgraphName,
-    ) -> Result<Option<SubgraphDeploymentId>, Error>;
+    ) -> Result<Option<SubgraphDeploymentId>, Error> {
+        // Find subgraph entity by name
+        let subgraph_entities = self
+            .find(SubgraphEntity::query().filter(EntityFilter::Equal(
+                "name".to_owned(),
+                name.to_string().into(),
+            )))
+            .map_err(QueryError::from)?;
+        let subgraph_entity = match subgraph_entities.len() {
+            0 => return Ok(None),
+            1 => {
+                let mut subgraph_entities = subgraph_entities;
+                Ok(subgraph_entities.pop().unwrap())
+            }
+            _ => Err(format_err!(
+                "Multiple subgraphs found with name {:?}",
+                name.to_string()
+            )),
+        }?;
+
+        // Get current active subgraph version ID
+        let current_version_id = match subgraph_entity
+            .get("currentVersion")
+            .ok_or_else(|| format_err!("Subgraph entity without `currentVersion`"))?
+        {
+            Value::String(s) => s.to_owned(),
+            Value::Null => return Ok(None),
+            _ => {
+                return Err(format_err!(
+                    "Subgraph entity has wrong type in `currentVersion`"
+                ));
+            }
+        };
+
+        // Read subgraph version entity
+        let version_entity_opt = self
+            .get(SubgraphVersionEntity::key(current_version_id))
+            .map_err(QueryError::from)?;
+        if version_entity_opt == None {
+            return Ok(None);
+        }
+        let version_entity = version_entity_opt.unwrap();
+
+        // Parse subgraph ID
+        let subgraph_id_str = version_entity
+            .get("deployment")
+            .ok_or_else(|| format_err!("SubgraphVersion entity without `deployment`"))?
+            .to_owned()
+            .as_string()
+            .ok_or_else(|| format_err!("SubgraphVersion entity has wrong type in `deployment`"))?;
+        SubgraphDeploymentId::new(subgraph_id_str)
+            .map_err(|()| {
+                format_err!("SubgraphVersion entity has invalid subgraph ID in `deployment`")
+            })
+            .map(Some)
+    }
+
+    /// Read all version entities pointing to the specified deployment IDs and determine whether
+    /// they are current or pending in order to produce `SubgraphVersionSummary`s.
+    fn read_subgraph_version_summaries(
```

Comment:
I found the previous assignment creation/deletion logic hard to follow, so I refactored to create this method and the following method (`reconcile_assignments`). This method finds what versions currently exist. `reconcile_assignments` takes "before" and "after" lists of what versions existed vs what versions will exist, diffs them, and handles creating/removing assignments.

These two methods are used anywhere assignments are created and/or removed, which is currently three places:

- `create_subgraph_version` (i.e. subgraph_deploy)
- `remove_subgraph`
-  BlockStream when setting the `synced` flag